### PR TITLE
E-989 feat: prevent 'regularPassword' from appearing in social login flow

### PR DESCRIPTION
**Pull request by ifiri**

### CODE REVIEW

**Comment by ifiri** [comment]
@dudiq ready for review
@kazurus signUp flow stay unchanged

File: src/components/screens/Authenticator/Authenticator.container.vue [hunk 1]
```diff
@@ -45,6 +45,11 @@ export default {
       type: Boolean,
```
Comment (dudiq) [inline src/components/screens/Authenticator/Authenticator.container.vue:45]: if i correct understand this task, you just need to remove this block of code
```
      if (SING_IN_CHALLENGE_TYPES.includes(this.challengeType)) {
        this.$emit('sign-in', { email });
        return;
      }
```
in `SignIn.interactor.vue`

and that's all

Comment (ifiri) [reply src/components/screens/Authenticator/Authenticator.container.vue:45]: we skip `nextScreen` call then

Comment (ifiri) [reply src/components/screens/Authenticator/Authenticator.container.vue:45]: current decision works and additionally provides way to detect was sign in by socials or not

Comment (dudiq) [reply src/components/screens/Authenticator/Authenticator.container.vue:45]: server side is not implemented.
we have a bug in the auth system?

Comment (ifiri) [reply src/components/screens/Authenticator/Authenticator.container.vue:45]: server side is in another tssk
current issue is about prevent `password` window from appearing when you sign in with socials

Comment (ifiri) [reply src/components/screens/Authenticator/Authenticator.container.vue:45]: issue-related bugs was fixed here

Comment (dudiq) [reply src/components/screens/Authenticator/Authenticator.container.vue:45]: server should implement first their fixes and then you need remove the code, what i've pasted.
@kazurus am i right? or this task about something other?

Comment (kazurus) [reply src/components/screens/Authenticator/Authenticator.container.vue:45]: @dudiq you're right. We should wait the server fix, and then only skip password forms for social + active otp

Comment (kazurus) [reply src/components/screens/Authenticator/Authenticator.container.vue:45]: But removing the code above doesn't solve the issue. If we have social + active otp, we should ask the code

Comment (dudiq) [reply src/components/screens/Authenticator/Authenticator.container.vue:45]: ok, understand

File: src/components/screens/Authenticator/modules/SignIn/SignIn.interactor.vue [hunk 2]
```diff
@@ -69,7 +69,7 @@ export default {
 
     async onSocial({ email }) {
       if (SING_IN_CHALLENGE_TYPES.includes(this.challengeType)) {
-        this.$emit('sign-in', { email });
+        this.$emit('sign-in-social', { email });
```
Comment (dudiq) [inline src/components/screens/Authenticator/modules/SignIn/SignIn.interactor.vue:72]: no need it to rename, just change to
`this.$emit('sign-in-social', { email, isSocial: true });` here
and in upper code
change 
`this.$emit('sign-in', { email, serverMode });` to 
`this.$emit('sign-in', { email, serverMode, isSocial: false });`

and for `SignUp.interactor.vue` change
`this.$emit('sign-up', { email, password, isSignUp: true });`
to
`this.$emit('sign-up', { email, password, isSignUp: true, isSocial: false });`

File: src/components/screens/Authenticator/modules/SignIn/SignIn.interface.vue [hunk 3]
```diff
@@ -4,6 +4,7 @@
     :is-public="isPublic"
     :challenge-type="challengeType"
     @sign-in="onSignIn"
+    @sign-in-social="onSignInSocial"
```
Comment (dudiq) [inline src/components/screens/Authenticator/modules/SignIn/SignIn.interface.vue:7]: and no need this event and above

File: src/components/screens/Authenticator/modules/SignIn/SignIn.interface.vue [hunk 4]
```diff
@@ -36,6 +37,10 @@ export default {
       this.$emit('complete', { email, serverMode, isSignUp });
     },
 
+    onSignInSocial({ email }) {
```
Comment (dudiq) [inline src/components/screens/Authenticator/modules/SignIn/SignIn.interface.vue:40]: no needed

File: src/components/screens/Authenticator/Authenticator.container.vue [hunk 5]
```diff
@@ -53,6 +58,19 @@ export default {
     },
   },
 
+  watch: {
```
Comment (dudiq) [inline src/components/screens/Authenticator/Authenticator.container.vue:61]: no needed too

**Comment by ifiri** [comment]
@dudiq done

**Comment by dudiq** [comment]
@lamartire ready for approve

**Comment by dudiq** [comment]
awaiting server implementation

**Comment by ifiri** [comment]
@kazurus not approved, additional fixes required. Always got "create password" window.

**Comment by lamartire** [comment]
@ifiri mention me, when it would be ready for approve

**Comment by ifiri** [comment]
@lamartire done, ready for approve

**Comment by dudiq** [comment]
sms code does not correct process in server:
- you should enable sms otp for your account in vault
- then logout
- and login with google
- when you will receive sms otp modal for, try to reload page
- you will be logged in vault, but you should not

---

and the next:
- you account should use sms otp mode
- when you will receive sms otp modal after login by google, try to enter code
- server return error
```
code: 406
error: "crypto/bcrypt: hashedPassword is not the hash of the given password"
message: "Not Acceptable"
```

File: src/components/screens/Authenticator/Authenticator.container.vue (hunk 1 above)
Comment (dudiq) [inline src/components/screens/Authenticator/Authenticator.container.vue:45]: sms code does not correct process in server:
- you should enable sms otp for your account in vault
- then logout
- and login with google
- when you will receive sms otp modal for, try to reload page
- you will be logged in vault, but you should not

---

and the next:
- you account should use sms otp mode
- when you will receive sms otp modal after login by google, try to enter code
- server return error
```
code: 406
error: "crypto/bcrypt: hashedPassword is not the hash of the given password"
message: "Not Acceptable"
```

**Comment by kazurus** [comment]
approved for bug representation

**Pull request merged**